### PR TITLE
Arista processor

### PR DESCRIPTION
**Pull request by h00die**
This PR adds processing of Arista devices/configs by adding a config processing library (and tests), a post module to gather the information while on device, and an aux module to process a config offline.

I'm up for suggestions on other fields to process. I have 0 experience with these devices, so I just kind of looked around at what may be of use.

You can download vEOS for GNS3 from the GNS3 marketplace, thats what I did to make a router and config and such.

I will add the post module shortly.

## Verification

- [ ] Make sure spec passes
~~- [ ] try the post module against a device~~
- [ ] try the aux module against a config (i provided mine in the docs)
- [ ] check documentation doesn't have issues

### CODE REVIEW

**Comment by h00die** [comment]
@SecurityBytesMe if you want to go ahead and try out the aux/admin module with the configs you have, let me know if there are any issues or additional fields you think I should pull in.

**Comment by h00die** [comment]
This block will need to be moved to the networking sub-header once #13860 lands
https://github.com/rapid7/metasploit-framework/pull/13870/commits/c370fac409c767edc460ee22540b6eb475a4a4b5#diff-d0c29526b6771d9377fb1d887827a1eaR24

I incremented the numbering on https://github.com/rapid7/metasploit-framework/pull/13870/files#diff-770fbdf33e9a5906fefbfd2482f9d4cbR80 to adjust for #13860

**Comment by h00die** [comment]
Also of note, as posted in slack, while the ssh_login module (and lib) can send input and get output, the `enum_arista` post module (not included in here) does NOT get input/output with either `sessions.shell_command` nor `cmd_exec`.

**Comment by SecurityBytesMe** [comment]
@h00die, Testing the module in my environment checks out good on my end!

**Comment by h00die** [comment]
Rebased to include the mikrotik changes, as well as now #13962 complaint.  I believe this is ready for testing.

File: modules/auxiliary/admin/networking/arista_config.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Arista
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Arista Configuration Importer',
+        'Description' => %q{
+          This module imports a Arista device configuration.
```
Comment (space-r7) [inline modules/auxiliary/admin/networking/arista_config.rb:17]:
```suggestion
          This module imports an Arista device configuration.
```

**Review by space-r7 — COMMENTED** [review]
Thanks for adding this! Just minor suggestions / questions.

File: lib/msf/core/auxiliary/arista.rb [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+          scomm = Regexp.last_match(1).strip
+          print_good("#{thost}:#{tport} SNMP Community (#{stype}): #{scomm}")
+
+          if framework.db.active
+            cred = credential_data.dup
+            if stype.downcase == 'ro'
+              cred[:access_level] = 'RO'
+            else
+              cred[:access_level] = 'RW'
+            end
```
[the remark this answers is not in the document]
Comment (h00die) [reply lib/msf/core/auxiliary/arista.rb:90]: copied from the cisco module, will fix them both on this PR.

File: modules/auxiliary/admin/networking/arista_config.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  end
+
+  def run
+    unless ::File.exist?(datastore['CONFIG'])
+      fail_with Failure::BadConfig, "Arista config file #{datastore['CONFIG']} does not exists!"
+    end
+    arista_config = ::File.open(datastore['CONFIG'], 'rb')
+    print_status('Importing config')
+    arista_eos_config_eater(datastore['RHOSTS'], datastore['RPORT'], arista_config.read)
+    print_good('Config import successful')
```
[the remark this answers is not in the document]
Comment (h00die) [reply modules/auxiliary/admin/networking/arista_config.rb:41]: that function has a host of print statements in it, so they should get back a lot of things from it directly

**Comment by space-r7** [comment]
Rebased to resolve the conflict in `lib/msf/core/auxiliary/cisco.rb`.

Tested module and spec:

```
msf6 > use auxiliary/admin/networking/arista_config 
msf6 auxiliary(admin/networking/arista_config) > set rhosts 172.16.215.136
rhosts => 172.16.215.136
msf6 auxiliary(admin/networking/arista_config) > set config /tmp/arista-conf
config => /tmp/arista-conf
msf6 auxiliary(admin/networking/arista_config) > run
[*] Running module against 172.16.215.136

[*] Importing config
[+] 172.16.215.136:22 Hostname: switch, Device: vEOS, OS: EOS, Version: 4.23.4.2M
[+] 172.16.215.136:22 AAA Username 'root' with Hash: $6$dVVGpxQASFh69zU1$jrSQxwjbuvkHYLTyTmVyAuCGSxoQ/4y6ZBk7LHm7F5yX1aqAxANLQEIU8.jZ8GPZK7ovNqmcoF/GIJV5XBBhk1
[+] 172.16.215.136:22 Username 'test-user' and Hash: $6$txiBHX9gLwsEZJIO$fsHDkoPAeLUCvCdmToHbOP2ewbohOecXH/1biIy0oV58pYzqQin7S1SBc7JLHIMA7IOIN8G5DuzvuqE/rzuto1
[+] Config import successful
[*] Auxiliary module execution completed
```

```
bundle exec rspec spec/lib/msf/core/auxiliary/arista_spec.rb
Run options: include {:focus=>true}

All examples were filtered out; ignoring {:focus=>true}

Randomized with seed 42042
Msf::Auxiliary::Arista ...........

Top 10 slowest examples (1.82 seconds, 97.5% of total time):
  Msf::Auxiliary::Arista deals with user details deals with no roles and md5 passwords
    0.45849 seconds ./spec/lib/msf/core/auxiliary/arista_spec.rb:185
  Msf::Auxiliary::Arista#create_credential_and_login associates the Metasploit::Credential::Core with a task if passed
    0.30893 seconds ./spec/lib/msf/core/auxiliary/arista_spec.rb:81
  Msf::Auxiliary::Arista deals with user details deals with no roles and plaintext passwords
    0.24348 seconds ./spec/lib/msf/core/auxiliary/arista_spec.rb:209
  Msf::Auxiliary::Arista#create_credential_and_login creates a Metasploit::Credential::Login
    0.22233 seconds ./spec/lib/msf/core/auxiliary/arista_spec.rb:78
  Msf::Auxiliary::Arista#arista_eos_config_eater deals with hostname
    0.13451 seconds ./spec/lib/msf/core/auxiliary/arista_spec.rb:100
  Msf::Auxiliary::Arista#arista_eos_config_eater deals with model and version
    0.12578 seconds ./spec/lib/msf/core/auxiliary/arista_spec.rb:92
  Msf::Auxiliary::Arista deals with user details deals with roles and sha512 passwords
    0.10544 seconds ./spec/lib/msf/core/auxiliary/arista_spec.rb:161
  Msf::Auxiliary::Arista#arista_eos_config_eater deals with aaa root secret
    0.09278 seconds ./spec/lib/msf/core/auxiliary/arista_spec.rb:132
  Msf::Auxiliary::Arista#arista_eos_config_eater deals with enable passwords
    0.08288 seconds ./spec/lib/msf/core/auxiliary/arista_spec.rb:108
  Msf::Auxiliary::Arista deals with SNMP details saves RW correct
    0.04416 seconds ./spec/lib/msf/core/auxiliary/arista_spec.rb:238

Finished in 1.86 seconds (files took 11.22 seconds to load)
11 examples, 0 failures
```

**Comment by space-r7** [comment]
## Release Notes

This adds a library for processing Arista switch configurations, an auxiliary module to import Arista switch config information to the database, and a spec for the library.

**Pull request merged**